envoy/http/header_map.h

```diff
@@ -231,6 +236,11 @@ class HeaderString {
 
   bool valid() const;
 
+  // Test only method that does not have validation and allows setting arbitrary values.
+  // Test code that needs access to it is declared below.
+  friend class Http2::Http2CodecImplTest;
+  void setCopyUnvalidated(absl::string_view view);
```

Comment:
Should we name this `setCopyUnvalidateForTest`? I think we've done this in other places to re-enforce the test-only status of the method.

Reply:
Done